feat: 5865 add data node v2 apis

guoguojin opened the pull request:
Closes #5685
Closes #5764

guoguojin committed:
feat: 5865 add data node v2 apis

jeremyletang commented on script/postgres-docker.sh:
```diff
@@ -0,0 +1,7 @@
+           -e POSTGRES_PASSWORD=vega \
+           -e POSTGRES_DB=vega \
+           -p 5432:5432 \
+           timescale/timescaledb:2.6.0-pg14
```
Is that the last version?

guoguojin replied:
No, that needs to be updated, I copied it from the old Data-Node project, will update

edd commented on datanode/entities/entities.go:
```diff
@@ -15,7 +15,7 @@ package entities
 
 type PagedEntity[T proto.Message] interface {
-	Market | Party | Trade | Order | MarginLevels | MarketData | Reward | Candle | Deposit | Withdrawal | Asset | OracleSpec | OracleData | Position | LiquidityProvision | Vote | Transfer | AccountBalance | Proposal | Delegation | Node | NetworkParameter | Checkpoint | StakeLinking
+	Market | Party | Trade | Order | MarginLevels | MarketData | Reward | Candle | Deposit | Withdrawal | Asset | OracleSpec | OracleData | Position | LiquidityProvision | Vote | Transfer | AccountBalance | Proposal | Delegation | Node | NetworkParameter | Checkpoint | StakeLinking | NodeSignature | KeyRotation | ERC20MultiSigSignerAddedEvent | ERC20MultiSigSignerRemovedEvent
```
**question (non-blocking)**:  I can see entities in here that are not reflected in the GraphQL schema. Could you check to see that all of them are? Specifically `ERC20MultiSigSignerAddedEvent` jumped out.

guoguojin replied:
There are a few functions that are exposed on the gRPC level that haven't made it through to the GraphQL side of things. So far in this migration, we have only migrated existing functionality of the GraphQL and gRPC functions.

We did have a discussion yesterday about this yesterday, it shouldn't take much work to plumb in the gRPC functions to the GraphQL API, but for now I've not added anything "new" to the GraphQL API, I've only deprecated functions that needed to be replaced with the newer cursor based paginated function.

I hope that makes sense.

edd replied:
It makes sense not to add new functionality, but we have recently had a few bugs that come about due to fields in the GRPC not being added/mapped/reflected properly in the GraphQL schema, resulting in the graphql server throwing errors (https://github.com/vegaprotocol/data-node/issues/594 for a recent example)

I guess I'm just asking for a sanity check that if these new fields are in GRPC, and I was subscribed in GraphQL to `busEvents`, and one of these was emitted, that it wouldn't break the subscription by emitting a server side error?

guoguojin replied:
Absolutely, there are some other plumbing that's missing in the GraphQL resolvers which I'm currently working through. I'll make sure everything is mapped correctly.

guoguojin committed:
chore: Merge branch 'develop' into feat/5685-add-data-node-v2-apis

guoguojin committed:
chore: clean up GraphQL resolvers

- Mark deprecated old v1 GQL resolvers
- Ensure non-deprecated GQL resolvers use v2 data-node gRPC API calls
- Close #5764
- Address review comments

pgs31 reviewed (dismissed):
It's a bit of a beast so I haven't been through with a fine toothed comb but it looks pretty sensible to me. I've got a PR incoming that simplifies the pagination stuff a bit but I've left the existing code in so it won't break

guoguojin committed:
chore: resolve merge conflicts with develop
